### ISSUE TITLE
The data shape of the triplet if the number of input functions of branch nets is more than one?

### ISSUE THREAD

**Issue by WangYicunZJU** [lululxvi/deeponet#9]
Hello:
First of all, thank you very much for your excellent work!
I have met a little problem.
How to reshape the dataset (or change the code) if the number of input functions of branch nets is more than one?
Just like G_phi in this paper("DeepM&Mnet: Inferring the electroconvection multiphysics fields..."), (c+, c-) -->phi.
Or like G_rhoNO in this paper("DeepM&Mnet for hypersonics..."), (rho_N2, rho_O2) --> rho_NO.
Specifically, what about the data shape of the triplet?
Thanks in advance.

**Comment by lululxvi** [comment]
Good question. In the DeepMMnet paper, we simply concatenate the two functions into a long vector.

**Comment by WangYicunZJU** [comment]
Hello Dr. Lu,
It is very nice to get your reply.
Following your answer, I think the data of the branch nets may be like this:
[ u1(x1), ..., u1(xm) ]   for one input function u1;
[ u1(x1), ..., u1(xm), u2(x1), ..., u2(xm) ]   for two input function u1 & u2.
Or like this:
[ u1(x1), u2(x1),..., u1(xm), u2(xm) ] ?
I think the former seems to be the case. Do I have the correct understanding?
I am looking forward to your reply.
Thanks in advance.

**Comment by lululxvi** [comment]
Yes, we use the former.

**Comment by WangYicunZJU** [comment]
OK, Dr. Lu.
I think I have got it. Thanks again!

**Comment by cfd-ai** [comment]
@lululxvi 
I have a similar question.
Imagine you have a transient simulation of a 3D domain and you are interested in an averaged quantity, e.g., Maximum Velocity  V(t) .  Then you would repeat this for various parameters (a parametric space), like Inlet Pressure (many Pi) and Inlet Temperature (many Ti).  To translate this to the DeepONet terminology, I guess:

G(u)(y) =  V(t)(Pi,Ti)
That is to say, 
u  is  t  ::  Input to Branch is time
(y) is (Pi, Ti)  ::  Input to Trunk is pair of (Pi, Ti)

Provided that the t array is always the same for all training, test, and future prediction values, I mean you are always interested in certain time values like [t1=0.1, t2=0.2, t3=0.3, ...], i.e., the objective is to predict V(t) always at the same times for any given (Pi,Ti), 
then I wonder which of the following forms is correct for the DeepONet Triple Data:

```python
X_train=
[
	[
		[ V(t1)(P1,T1), V(t2)(P1,T1), V(t3)(P1,T1)... ] ,
		[ V(t1)(P2,T2), V(t2)(P2,T2), V(t3)(P2,T2)... ] ,
		...
	],
	[
		[ t1, t2, t3... ] ,
		[ t1, t2, t3... ] ,
		...
	]
]

y_train=
[
	[P1 , T1] ,
	[P2 , T2] ,
	...
]
```


OR


```python
X_train=
[
	[
		[ V(t1)(P1,T1) ] ,
		[ V(t2)(P1,T1) ] ,
		[ V(t3)(P1,T1) ] ,
		...
		[ V(t1)(P2,T2) ] ,
		[ V(t2)(P2,T2) ] ,
		[ V(t3)(P2,T2) ] ,
		...
	],
	[
		[t1] ,
		[t2] ,
		[t3] ,
		...
		[t1] ,
		[t2] ,
		[t3] ,
		...
	]
]

y_train=
[
	[P1 , T1] ,
	[P1 , T1] ,
	[P1 , T1] ,
	...
	[P2 , T2] ,
	[P2 , T2] ,
	[P2 , T2] ,
	...
]
```

Basically, train & predict over the entire [t] array together?  or train & predict per each t value?
or both of them are totally wrong?
Thanks!

**Comment by lululxvi** [comment]
I am a little confused about the setup. Let us consider a simple: G(u)(t) = s(t), i.e., for any u(t), I want to predict the corresponding s(t). Then t is the input to trunk net, and u(t) is the input to branch net.
